autogpt_platform/backend/backend/sdk/builder.py

```diff
@@ -90,6 +90,27 @@ def with_api_key_from_settings(
             )
         return self
 
+    def with_user_password(
+        self, username_env_var: str, password_env_var: str, title: str
+    ) -> "ProviderBuilder":
+        """Add username/password support with environment variable names."""
+        self._supported_auth_types.add("user_password")
+
+        # Check if credentials exist in environment
+        username = os.getenv(username_env_var)
+        password = os.getenv(password_env_var)
+        if username and password:
+            self._default_credentials.append(
+                UserPasswordCredentials(
+                    id=f"{self.name}-default",
+                    provider=self.name,
+                    username=SecretStr(username),
+                    password=SecretStr(password),
+                    title=title,
+                )
+            )
+        return self
+
```

Comment:
These should probably be optional

Reply:
ex: possible to define a user/pass requireent without providing default based on the types either via an overload or default none